app/Jobs/BroadcastNotification.php

```diff
@@ -164,17 +186,35 @@ private function onBeatmapsetDiscussionQualifiedProblem()
             ->get();
 
         foreach ($notificationOptions as $notificationOption) {
-            if (count(array_intersect($notificationOption->details['modes'], $modes)) > 0) {
+            if (count(array_intersect($notificationOption->details['modes'] ?? [], $modes)) > 0) {
                 $this->receiverIds[] = $notificationOption->user_id;
             }
         }
 
+        $this->receiverIds = static::filterBeatmapsetModdingUserIds($this->receiverIds);
+
         $this->assignBeatmapsetDiscussionNotificationDetails();
     }
 
     private function onBeatmapsetDisqualify()
     {
-        $this->receiverIds = static::beatmapsetReceiverIds($this->object);
+        $modes = $this->object->playmodes()->all();
+        $modes = array_map(function ($modeInt) {
+            return Beatmap::modeStr($modeInt);
+        }, $modes);
+
+        $notificationOptions = UserNotificationOption
+            ::where(['name' => Notification::BEATMAPSET_DISQUALIFY])
+            ->whereNotNull('details')
+            ->get();
+
+        $this->receiverIds = static::beatmapsetWatcherUserIds($this->object);
+
+        foreach ($notificationOptions as $notificationOption) {
+            if (count(array_intersect($notificationOption->details['modes'] ?? [], $modes)) > 0) {
```

Comment:
should this check `push` option as well? Considering it's part of `supportsNotifications` check.

Although there isn't any way to set it from frontend. Or maybe should check the one in `BEATMAPSET_MODDING`...? :thinking:

Reply:
No, this option doesn't have anything to do with delivery mode